File: scytale-cipher/enc.cpp
#include <iostream>
#include <string>
#include <random>

using namespace std;


char random_alphabet();  // return one random alphabet
string encode(int key, string text);


int main() {
    cout << "=== scytale-cipher ===" << endl;

    int key;
    cout << "Input key: ";
    cin >> key;
    cin.ignore();

    string text;
    cout << "Input plain text: ";
    getline(cin, text);
    text.erase(remove(text.begin(), text.end(), ' '), text.end());
    cout << text << "\n";
    string cipher = encode(key, text);
    cout << cipher << "\n";
    return 0;
}


char random_alphabet() {
    random_device rnd;
    mt19937 mt(rnd());
    uniform_int_distribution<> rand(0, 25);

    int shift = rand(mt);
    return 'a' + shift;
}


string encode(int key, string text) {
    
    int text_length = text.size();
    int reminder = text_length % key;
    int quotient = text_length / key;
    if (reminder != 0) {
        quotient++;
        for (int i=0; i<(key - reminder); i++) {
            text += random_alphabet();
        }
    }
    
    string cipher;
    for (int i=0;i<key; i++) {
        for (int j=0; j<quotient; j++){
            cipher += text[j*key + i];
        }
    }

    return cipher;
}

File: scytale-cipher/enc.py
import random

def random_alphabet():
    """
    === return ===
    random alphabet in lower cases
    """

    alphabet_list = [chr(i + ord('a')) for i in range(26)]
    return random.choice(alphabet_list)


def enc(text, key):
    # scytale encryption
    text_list = [c for c in text]

    len_text = len(text_list)
    quotient_key = -(-len_text // key)
    mod_key = len_text % key

    for i in range((key - mod_key)%key):
        text_list.append(random_alphabet())

    cipher = ""
    for i in range(key):
        for j in range(quotient_key):
            cipher += text_list[key*j + i]
    return cipher


def main():
    key = int(input("input number(int)\n"))
    text = input("input plain text\n")
    print(enc(text, key))

if __name__ == "__main__":
    main()
